frontend/src/features/settings/components/creditAccount/AddUserModal.tsx

```diff
@@ -0,0 +1,183 @@
+import { Button, Dialog } from "@mui/material";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPlusCircle, faInfoCircle } from "@fortawesome/free-solid-svg-icons";
+import { FormProvider, useForm } from "react-hook-form";
+import {
+  useAddCreditAccountUserMutation,
+  useGetCreditAccountQuery,
+} from "../../hooks/creditAccount";
+import {
+  CREDIT_ACCOUNT_USER_TYPE,
+  CreditAccountUser,
+} from "../../types/creditAccount";
+import { CompanyProfile } from "../../../manageProfile/types/manageProfile";
+import { StatusChip } from "./StatusChip";
+import "./AddUserModal.scss";
+import { useContext } from "react";
+import OnRouteBCContext from "../../../../common/authentication/OnRouteBCContext";
+import { getDefaultRequiredVal } from "../../../../common/helpers/util";
+
+export const AddUserModal = ({
+  showModal,
+  onCancel,
+  onConfirm,
+  userData,
+}: {
+  showModal: boolean;
+  onCancel: () => void;
+  onConfirm: () => void;
+  userData: CompanyProfile;
+}) => {
+  const { companyId } = useContext(OnRouteBCContext);
+
+  const formMethods = useForm<{ comment: string }>({
+    reValidateMode: "onChange",
+  });
+
+  const { handleSubmit } = formMethods;
+
+  const isActionSuccessful = (status: number) => {
+    return status === 200;
+  };
+
+  const { data: creditAccount } = useGetCreditAccountQuery(
+    getDefaultRequiredVal(0, companyId),
+  );
+  const { data: userCreditAccount } = useGetCreditAccountQuery(
+    userData.companyId,
+  );
+
+  /** TODO Currently this boolean will prevent companies who are previous credit account holders
+   * from being added as users. Once the API allows the creation of credit accounts for companies other than
+   * Parisian Trucking we can implement a check to determine if the previously associated credit account of the
+   * ex holding company is "CLOSED" and allow the adding of this company as a user to the credit account
+   */
+  const existingCreditAccountHolder =
+    userCreditAccount?.creditAccountUsers?.find(
+      (user: CreditAccountUser) =>
+        user.userType === CREDIT_ACCOUNT_USER_TYPE.HOLDER,
+    );
+
+  const { mutateAsync, isPending } = useAddCreditAccountUserMutation();
+
+  const handleAddUser = async () => {
+    if (creditAccount?.creditAccountId) {
+      const { status } = await mutateAsync({
+        companyId: getDefaultRequiredVal(0, companyId),
+        creditAccountId: creditAccount.creditAccountId,
+        userData,
+      });
+
+      if (isActionSuccessful(status)) {
+        onConfirm();
+      }
+    }
```

Comment:
Do something to handle the error condition, either console.error the error message, or navigate to unexpected error page. Don't leave it and do nothing for the error.

Reply:
Similarly here, don't print the status, but instead print something more meaningful (eg. "Add user failed" or something).